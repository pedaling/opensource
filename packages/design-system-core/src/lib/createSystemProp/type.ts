import type { Theme } from '../types';

export type InterpolationFn = (props: Record<string, any>) => Record<string, any>;

export type SystemPropConfig = {
  property: string;
  styleProperty?: string;
  scale?: keyof Pick<Theme, 'colors' | 'space' | 'opacity'>;
  disabled?: boolean;
  transform?: (value: any) => Record<string, any> | undefined;
};

export type SystemProp = {
  (props: Record<string, any>): Record<string, any>;
  propName: string;
  disabled: boolean;
};
