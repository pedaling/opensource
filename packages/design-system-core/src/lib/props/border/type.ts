import type { ResponsiveValue } from '../../types';

export type BorderProps = {
  borderWidth?: ResponsiveValue<number>;
  borderStyle?: ResponsiveValue<'solid' | 'none'>;
  borderColor?: ResponsiveValue<string>;
  borderTopWidth?: ResponsiveValue<number>;
  borderTopStyle?: ResponsiveValue<'solid' | 'none'>;
  borderTopColor?: ResponsiveValue<string>;
  borderRightWidth?: ResponsiveValue<number>;
  borderRightStyle?: ResponsiveValue<'solid' | 'none'>;
  borderRightColor?: ResponsiveValue<string>;
  borderBottomWidth?: ResponsiveValue<number>;
  borderBottomStyle?: ResponsiveValue<'solid' | 'none'>;
  borderBottomColor?: ResponsiveValue<string>;
  borderLeftWidth?: ResponsiveValue<number>;
  borderLeftStyle?: ResponsiveValue<'solid' | 'none'>;
  borderLeftColor?: ResponsiveValue<string>;
  borderRadius?: ResponsiveValue<number | string>;
  borderTopLeftRadius?: ResponsiveValue<number | string>;
  borderTopRightRadius?: ResponsiveValue<number | string>;
  borderBottomLeftRadius?: ResponsiveValue<number | string>;
  borderBottomRightRadius?: ResponsiveValue<number | string>;
  outlineWidth?: ResponsiveValue<number>;
  outlineStyle?: ResponsiveValue<'solid' | 'none'>;
  outlineColor?: ResponsiveValue<string>;
  outlineOffset?: ResponsiveValue<number>;
};
