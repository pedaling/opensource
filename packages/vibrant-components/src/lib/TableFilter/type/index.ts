export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
