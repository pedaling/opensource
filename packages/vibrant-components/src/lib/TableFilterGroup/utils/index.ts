import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
