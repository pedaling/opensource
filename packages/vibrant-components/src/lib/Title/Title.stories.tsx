import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { Title } from './Title';

export default {
  title: 'Title',
  component: Title,
  args: {
    level: 1,
    children: 'Sample Text',
  },
} as ComponentMeta<typeof Title>;

export const Basic: ComponentStory<typeof Title> = ({ children, ...restProps }) => (
  <Title {...restProps}>{children}</Title>
);
