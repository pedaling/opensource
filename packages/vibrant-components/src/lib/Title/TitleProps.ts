import type { PositionSystemProps, ResponsiveValue, TextProps, TypographySystemProps } from '@vibrant-ui/core';
import { propVariant, withVariation } from '@vibrant-ui/core';
import type { TypographyWeight } from '@vibrant-ui/theme';

export type TitleProps = Omit<TextProps, keyof PositionSystemProps | keyof TypographySystemProps> & {
  level: ResponsiveValue<1 | 2 | 3 | 4 | 5 | 6 | 7>;
  weight?: ResponsiveValue<TypographyWeight>;
};

export const withTitleVariation = withVariation<TitleProps>('Title')(
  propVariant({
    props: [
      {
        name: 'level',
        responsive: true,
      },
    ],
    variants: ({ level }) => ({
      typography: `title${level}` as const,
    }),
  }),
  propVariant({
    props: [
      {
        name: 'weight',
        responsive: true,
        default: 'bold' as const,
      },
    ],
    variants: ({ weight }) => ({
      fontWeight: weight,
    }),
  })
);
