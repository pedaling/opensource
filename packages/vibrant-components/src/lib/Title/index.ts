export { Title } from './Title';
export type { TitleProps } from './TitleProps';
