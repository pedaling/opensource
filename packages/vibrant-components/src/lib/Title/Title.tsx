import { Text } from '@vibrant-ui/core';
import { withTitleVariation } from './TitleProps';

export const Title = withTitleVariation(({ innerRef, ...restProps }) => <Text ref={innerRef} {...restProps} />);
