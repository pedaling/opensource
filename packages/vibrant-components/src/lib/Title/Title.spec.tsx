import { baseTheme } from '@vibrant-ui/theme';
import type { ReactRenderer } from '@vibrant-ui/utils/testing-web';
import { createReactRenderer } from '@vibrant-ui/utils/testing-web';
import { Title } from './Title';

describe('<Title />', () => {
  const { render } = createReactRenderer();

  let renderer: ReactRenderer;

  let element: HTMLElement;

  describe('when Title with level 3 rendered', () => {
    beforeEach(async () => {
      renderer = render(
        <Title data-testid="title" level={3}>
          Hello World
        </Title>
      );

      element = await renderer.findByTestId('title');
    });

    it('title3 style applied', () => {
      const themeTypography = baseTheme.typography;
      const themeTypographyWeight = baseTheme.typographyWeight;

      expect(element).toHaveStyleRule('font-size', themeTypography.title3.fontSize);

      expect(element).toHaveStyleRule(
        'font-weight',
        themeTypographyWeight[themeTypography.title3.fontWeight].fontWeight.toString()
      );

      expect(element).toHaveStyleRule('line-height', themeTypography.title3.lineHeight);
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });
});
