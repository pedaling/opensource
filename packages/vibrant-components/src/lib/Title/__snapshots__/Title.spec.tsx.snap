// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> when Title with level 3 rendered match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  font-family: Pretendard;
  color: #0C0C0C;
  font-size: 1.625rem;
  font-weight: 700;
  line-height: 2.25rem;
}

<div>
  <span
    class="emotion-0"
    data-testid="title"
  >
    Hello World
  </span>
</div>
`;
